flink-kubernetes/src/main/java/org/apache/flink/kubernetes/executors/KubernetesJobClusterExecutor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.executors;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.client.deployment.AbstractJobClusterExecutor;
+import org.apache.flink.core.execution.Executor;
+import org.apache.flink.kubernetes.KubernetesClusterClientFactory;
+
+/**
+ * The {@link Executor} to be used when executing a job in isolation.
+ * This executor will start a cluster specifically for the job at hand and
+ * tear it down when the job is finished either successfully or due to an error.
+ */
+@Internal
+public class KubernetesJobClusterExecutor extends AbstractJobClusterExecutor<String, KubernetesClusterClientFactory> {
```

Comment:
Since we don't support per-job mode in k8s in this pass, I don't think it makes any sense to introduce such executor.

Reply:
I have created a following PR to support per-job mode by adding `kubernetes-job.sh`. The user jar have been built in the docker image and `ClasspathJobGraphRetriever` will be used to get the job graph.
The problem is whether we have enough time to review and merge.

Reply:
I will remove the the job related class.